### ISSUE TITLE
Updating Non selected Content 

### ISSUE THREAD

**Issue by healthray** [microsoft/vscode-copilot-release#3553]
Type: <b>Bug</b>

even though I am asking it to update few selected lines , it still goes for other content written in file this behaviour is completely disasterous and ends up wreaking existing code

Extension version: 0.23.2
VS Code version: Code 1.96.1 (42b266171e51a016313f47d0c48aca9295b9cbb2, 2024-12-17T17:50:05.206Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7400 CPU @ 3.00GHz (4 x 3000)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.89GB (4.79GB free)|
|Process Argv|--crash-reporter-id f3078cff-f131-46ab-a151-90bb2bd478c0|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythonnoceb:30805159
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
dvdeprecation:31068756
dwnewjupyter:31046869
nativerepl1:31139838
pythonrstrctxt:31112756
nativeloc1:31192215
cf971741:31144450
iacca1:31171482
notype1:31157159
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530
6074i472:31201624

```

</details>

**Comment by salameer** [comment]
What I usually try to do to avoid this is to specify the exact lines of code that I'm looking to update. did you try doing that?

**Comment by healthray** [comment]
Yes, I also tried it by line number, giving it a specific function I wanted it to work on, but I am still having problems.

**Comment by salameer** [comment]
I dont know if you tried the following but it maybe worth trying :) 

Selected a number of lines with you mouse -> right click -> copilot -> add to chat.

Then you'll see in the chat box that it's added focus on the highlighted lines. Then ask it to update

**Comment by healthray** [comment]
It works sometimes but still some times working on big files When I ask it to tend one small block of selected block of code, it updates that part and ends up messing my code and removing or updating part, which is unwanted behaviour 

Also sometimes it updates small parts of code but removes the code which follows and adds something similar to below; here in that case, I had only asked it to check and update the pattern check if necessary for mirrorDeleteFileThreshold , I had mentioned that specifically in prompt as well

     mirrorDeleteFileThreshold: Joi.string().optional().pattern(/^\d+$/).allow("").description("Mirror Delete File Threshold"),
      mirrorDeleteRetentionEnabled: Joi.string().optional(),
      mirrorDeleteRetentionDays: Joi.string().optional(),
      sourceDeleteAfterRetentionEnabled: Joi.string().optional(),
      sourceDeleteAfterRetentionDays: Joi.string().optional(),
      skipExists: Joi.string().optional(),
      duplicateFolders: Joi.array().items(
        Joi.object({
        depth: Joi.number(),
        path: Joi.string(),
        })
      ).optional(),
      }),
    },
    // ... (rest of the cases remain unchanged)
    ],
  }).required(),
  }).label("UpdateProjectOptions");
